src/storage/storage.js

```
@@ -1,96 +1,159 @@
+ * @param {*} result Result from a createNamespace call
+ */
+function isDuplicateNamespaceError(result) {
+  return (
+    result &&
+    result.success == false &&
+    result.errors.length == 1 &&
+    result.errors[0].code == 10014
+  );
+}
+
+/**
+ * Attempts to create a namespace for the account
+ * @param {*} accountId
+ * @param {*} name
+ */
+async function createNamespace({ accountId, name }) {
+  accountId = accountId || process.env.CLOUDFLARE_ACCOUNT_ID;
+
+  if (!accountId) {
+    throw 'You must provide an account ID';
+  }
 
-    return result;
-  },
+  const result = await api.cfApiCall({
+    url: `/accounts/${accountId}/storage/kv/namespaces`,
+    method: `POST`,
+    contentType: `application/json`,
+    body: JSON.stringify({
+      title: name
+    })
+  });
+
+  return result;
+}
+
+/**
+ * Gets namespaces associated with the account
+ * @param {*} options
+ */
+async function getNamespaces({ accountId } = {}) {
+  accountId = accountId || process.env.CLOUDFLARE_ACCOUNT_ID;
+
+  if (!accountId) {
+    throw 'You must provide an account ID';
+  }
 
-  /**
-   * Gets namespaces associated with the account
-   * @param {*} options
-   */
-  async getNamespaces({accountId} = {}) {
-    accountId = accountId || process.env.CLOUDFLARE_ACCOUNT_ID;
+  let result = await api.cfApiCall({
+    url: `/accounts/${accountId}/storage/kv/namespaces`,
+    method: 'GET'
+  });
 
-    if (!accountId) {
-      throw("You must provide an account ID");
-    }
+  if (result.success && result.result) {
+    result = result.result;
+  }
 
-    let result = await api.cfApiCall({
-      url: `/accounts/${accountId}/storage/kv/namespaces`,
-      method: 'GET'
+  return result;
+}
+
+/**
+ * Get a namespace id by name
+ * @param {*} options
+ */
+async function getNamespaceByName({ accountId, name }) {
+  const account = accountId || process.env.CLOUDFLARE_ACCOUNT_ID;
+  let namespaceId = '';
+  if (!isUUID(name)) {
```

If you are checking for a UUID, you should throw an error or just remove this check entirely.